src/einsteinpy/geodesic.py

```diff
@@ -1,68 +1,212 @@
-            self.body.name,
-            self.metric.name,
-            self.body.parent.name,
-            self.body.parent.mass,
-        )
+        return f"Geodesic:\n\nMetric = ({self.metric}),\
+            \n\ninit_vec = ({self.init_vec}),\
+            \n\nTrajectory = ({self.trajectory})"
 
     def __str__(self):
-        return "Geodesic object body name= ({0}) , metric=({1}) , parent name=({2}) , parent mass=({3})".format(
-            self.body.name,
-            self.metric.name,
-            self.body.parent.name,
-            self.body.parent.mass,
-        )
+        return f"Geodesic:\n\nMetric = ({self.metric}),\
+            \n\ninit_vec = ({self.init_vec}),\
+            \n\nTrajectory = ({self.trajectory})"
 
     @property
     def trajectory(self):
         return self._trajectory
+
+    def calculate_trajectory(
+        self,
+        end_lambda=10.0,
+        stop_on_singularity=True,
+        OdeMethodKwargs={"stepsize": 1e-3},
+        return_cartesian=True,
+    ):
+        """
+        Calculate trajectory in spacetime, according to Geodesic Equations
+
+        Parameters
+        ----------
+        end_lambda : float
+            Affine Parameter, Lambda, where iterations will stop
+            Equivalent to Proper Time for Timelike Geodesics
+            Defaults to ``10``
+        stop_on_singularity : bool
+            Whether to stop further computation on reaching singularity
+            Defaults to ``True``
+        OdeMethodKwargs : dict
+            Kwargs to be supplied to the ODESolver
+            Dictionary with key 'stepsize' along with a float value is expected
+            Defaults to ``{'stepsize': 1e-3}``
+        return_cartesian : bool
+            Whether to return calculated values in Cartesian Coordinates
+            Defaults to ``True``
+
+        Returns
+        -------
+        ~numpy.ndarray
+            N-element array containing Lambda, where the geodesic equations were evaluated
+        ~numpy.ndarray
+            (n,8) shape array containing [x0, x1, x2, x3, v0, v1, v2, v3] for each Lambda
+
+        """
+        ODE = RK45(
+            fun=self.metric.f_vec,
+            t0=0.0,
+            y0=self.init_vec,
+            t_bound=end_lambda,
+            **OdeMethodKwargs,
+        )
+
+        vecs = list()
+        lambdas = list()
+        crossed_event_horizon = False
+        _scr = self.metric.sch_rad * 1.001
+
+        while ODE.t < end_lambda:
+            vecs.append(ODE.y)
+            lambdas.append(ODE.t)
+            ODE.step()
+            if (not crossed_event_horizon) and (ODE.y[1] <= _scr):
+                warnings.warn(
+                    "Test particle has reached Schwarzchild Radius. ", RuntimeWarning
+                )
+                if stop_on_singularity:
+                    break
+                else:
+                    crossed_event_horizon = True
+
+        vecs, lambdas = np.array(vecs), np.array(lambdas)
+
+        if not return_cartesian:
```

Comment:
Referencing #113.

This if-elif section (L125 - L146 & L190 - L203) is causing the cyclomatic complexity to go up. After the PR on `coordinates` is merged, I think, we can use coordinate classes, instead of strings, like "S" or "BL" or "KS". So, this whole section would be something like:
```
if not return_cartesian:
    return lambdas, vecs

cart_vecs = list()
for v in vecs:
    si_vals =  self.metric.coords(v[1], v[2], v[3], v[5], v[6], v[7]).convert_cartesian()
    cart_vecs.append(np.hstack((v[0], si_vals[:3], v[4], si_vals[3:])))

return lambdas, np.array(cart_vecs)
```

I would like to get your suggestions regarding this.

Reply:
Added @rishi-s8's suggestion and a fix detailed in [this](https://github.com/einsteinpy/einsteinpy/pull/512#issuecomment-647451480) comment. Issue resolved.

Reply:
For now, I have applied @rishi-s8's suggestions and some other changes, as discussed in [this](https://github.com/einsteinpy/einsteinpy/pull/512#issuecomment-647451480) comment. More changes are expected in the next PRs (after coordinates, though). Marking as resolved.